tasmota/tasmota_xdrv_driver/xdrv_90_esp32_dingtian_relay.ino

```diff
@@ -83,31 +91,31 @@ uint32_t DingtianReadWrite(uint32_t outputs)
  * Driver initialisation
  */
 
-#define DINGTIAN_SET_OUTPUT(pin,value)  { pinMode((pin), OUTPUT); digitalWrite((pin), (value)); }
-#define DINGTIAN_SET_INPUT(pin)         { pinMode((pin), INPUT); }
-
 void DingtianInit(void) {
   if (PinUsed(GPIO_DINGTIAN_CLK, GPIO_ANY) && PinUsed(GPIO_DINGTIAN_SDI) && PinUsed(GPIO_DINGTIAN_Q7)
-   && PinUsed(GPIO_DINGTIAN_PL) && PinUsed(GPIO_DINGTIAN_RCK)) {
+   && (PinUsed(GPIO_DINGTIAN_PL) || PinUsed(GPIO_DINGTIAN_OE)) && PinUsed(GPIO_DINGTIAN_RCK)) {
     // allocate Dingtian data structure
     Dingtian = (struct DINGTIAN_DATA*)calloc(1, sizeof(struct DINGTIAN_DATA));
     if (Dingtian) {
       // get pins
-      Dingtian->pin_clk = Pin(GPIO_DINGTIAN_CLK, GPIO_ANY);   // shift clock  : 595's SCLK & 165's CLK
-      Dingtian->pin_sdi = Pin(GPIO_DINGTIAN_SDI);             // Serial out   : 595's SER
-      Dingtian->pin_q7  = Pin(GPIO_DINGTIAN_Q7);              // Serial in    : 165's Q7
-      Dingtian->pin_pl  = Pin(GPIO_DINGTIAN_PL);              // Input load   : 595's nOE  & 165's PL (or SH/LD on some datasheet)
-      Dingtian->pin_rck = Pin(GPIO_DINGTIAN_RCK);             // Output load  : 595's RCLK & 165's CLKINH
+      Dingtian->pin_clk = Pin(GPIO_DINGTIAN_CLK, GPIO_ANY);                     // shift clock   : 595's SCLK & 165's CLK
+      Dingtian->pin_sdi = Pin(GPIO_DINGTIAN_SDI);                               // Serial out    : 595's SER
+      Dingtian->pin_q7  = Pin(GPIO_DINGTIAN_Q7);                                // Serial in     : 165's Q7
+      if (PinUsed(GPIO_DINGTIAN_PL)) Dingtian->pin_pl  = Pin(GPIO_DINGTIAN_PL); // Input load    : 595's nOE  & 165's PL (or SH/LD on some datasheet)
+      if (PinUsed(GPIO_DINGTIAN_OE)) Dingtian->pin_oe  = Pin(GPIO_DINGTIAN_OE); // Output enable : 595's nOE (v3.6.10)
```

Comment:
PL should be mandatory, OE optional
If OE not defined, a default value wouldbe in pin_oe to say "not used" to optimize above test

Reply:
PL is not mandatory for the new board.

OE or PL is mandatory.